### PR TITLE
feat(secrets): support git history scan in multiline parsers

### PR DESCRIPTION
**Pull request by lirshindalman**
support git history scan in multiline parsers

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by pazbechor — APPROVED** [review]

File: checkov/main.py [hunk 1]
```diff
@@ -667,8 +667,8 @@ def run(self, banner: str = checkov_banner) -> int | None:
                 repo_root_for_plan_enrichment=self.config.repo_root_for_plan_enrichment,
                 resource_attr_to_omit=self.config.mask,
                 # TODO modify the output for git_history secret and remove the rewrite of enable_git_history_secret_scan
-                # enable_git_history_secret_scan=self.config.scan_secrets_history,
-                enable_git_history_secret_scan=False,  # expose after unite git history with secret scan
+                enable_git_history_secret_scan=self.config.scan_secrets_history,
```
Comment (omryMen) [inline checkov/main.py:670]: I'm not sure we want to expose it yet..

File: checkov/secrets/plugins/detector_utils.py [hunk 2]
```diff
@@ -185,7 +185,8 @@ def extract_from_string(pattern: dict[Pattern[str], int] | None, string: str) ->
     for value_regex, group_number in pattern.items():
         match = value_regex.search(string)
         if match:
-            matches |= {match.group(group_number)}
+            remove_n = -1 if match.group(group_number).endswith('\n') else None
+            matches |= {match.group(group_number)[:remove_n]}
```
Comment (Eliran-Turgeman) [inline checkov/secrets/plugins/detector_utils.py:189]: you can use `rstrip` to strip characters from the right side of the string
```suggestion
            matches |= {match.group(group_number).rstrip('\n')}
```

File: checkov/secrets/scan_git_history.py [hunk 3]
```diff
@@ -105,7 +105,8 @@ def _get_commits_diff(self) -> Dict[str, Dict[str, str | Dict[str, str]]]:
                 base_diff_format = f'diff --git a/{file_diff.a_path} b/{file_diff.b_path}' \
                                    f'\nindex 0000..0000 0000\n--- a/{file_diff.a_path}\n+++ b/{file_diff.b_path}\n'
                 commits_diff.setdefault(current_commit_hash, {})
-                commits_diff[current_commit_hash][file_diff.a_path] = base_diff_format + file_diff.diff.decode()
+                file_name = file_diff.a_path if file_diff.a_path else file_diff.b_path
```
Comment (Eliran-Turgeman) [inline checkov/secrets/scan_git_history.py:108]: is `file_diff.b_path` guaranteed not to be None here?

Comment (lirshindalman) [reply checkov/secrets/scan_git_history.py:108]: if the a_path is None so yes

**Review by arielkru — APPROVED** [review]

**Pull request merged**